x-pack/plugins/security/server/lib/authorization/check_privileges.js

```diff
@@ -86,10 +80,7 @@ export function checkPrivilegesWithRequestFactory(server) {
         };
       }
 
-      if (!applicationPrivilegesCheck.privileges[loginAction] && await hasPrivilegesOnKibanaIndex()) {
-        const msg = 'Relying on index privileges is deprecated and will be removed in Kibana 7.0';
-        server.log(['warning', 'deprecated', 'security'], msg);
-
+      if (!applicationPrivilegesCheck.privileges[actions.login] && await hasPrivilegesOnKibanaIndex()) {
```

Comment:
I'm not sure if it's worth it or not, but now that we don't have to log the deprecation warning on a per-request basis, we could potentially do the entire privilege check in a single round-trip. Something like:

```javascript
const privilegeCheck = await callWithRequest(request, 'shield.hasPrivileges', {
 body: {
   index: [{
     names: [kibanaIndex],
     privileges: ['create', 'delete', 'read', 'view_index_metadata']
   }],
   applications: [{
     application,
     resources: [DEFAULT_RESOURCE],
     privileges
   }]
 }
});
```

A couple of thoughts on this approach:
1. We wouldn't be able to rely on ES's `has_all_requested`, but it should also be fairly trivial to derive that ourselves based on the response.
2. Non-legacy users will incur an additional overhead while ES checks the index privileges, which we will never use for them. This is likely negligible though?

Reply:
I like it... I'll make it so.